vultr/resource_reverse_name.go

```diff
@@ -0,0 +1,182 @@
+package vultr
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceReverseName() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceReverseNameCreate,
+		Read:   resourceReverseNameRead,
+		Update: resourceReverseNameUpdate,
+		Delete: resourceReverseNameDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"ip": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return net.ParseIP(old).Equal(net.ParseIP(new))
+				},
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceReverseNameCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Client)
+
+	id := d.Get("instance_id").(string)
+	ip := net.ParseIP(d.Get("ip").(string))
+	name := d.Get("name").(string)
+
+	log.Printf("[INFO] Setting reverse name for ip (%s)", ip)
+
+	setOrUpdateReverseName(client, id, ip, name)
+
+	d.SetId(fmt.Sprintf("%s/%s", id, ip))
+
+	return resourceReverseNameRead(d, meta)
+}
+
+func resourceReverseNameRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Client)
+
+	id, ip, err := splitID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.Set("instance_id", id)
+	d.Set("ip", ip.String())
+
+	if isIPv4(ip) {
+		ipv4s, err := client.ListIPv4(id)
+		if err != nil {
+			return fmt.Errorf("Error getting a list of IPv4 addresses for the instance (%s): %v", id, err)
+		}
+
+		for _, ipv4 := range ipv4s {
+			if ip.Equal(net.ParseIP(ipv4.IP)) {
+				d.Set("name", ipv4.ReverseDNS)
+				return nil
+			}
+		}
+
+		log.Printf("Can't get reverse name of IPv4 (%s) on instance (%s). Assuming that this IPv4 was deleted.", ip, id)
+		d.SetId("")
+		return nil
+	}
+
+	ipv6s, err := client.ListIPv6ReverseDNS(id)
+	if err != nil {
+		return fmt.Errorf("Error getting a list of IPv6 addresses for the instance (%s): %v", id, err)
+	}
+
+	for _, ipv6 := range ipv6s {
+		if ip.Equal(net.ParseIP(ipv6.IP)) {
+			d.Set("name", ipv6.ReverseDNS)
+			return nil
+		}
+	}
+
+	log.Printf("Can't get reverse name of IPv6 (%s) on instance (%s). Assuming that this IPv6 was deleted.", ip, id)
```

Comment:
In general this provider prefixes log statements with the log level. Can you add that here?